### ISSUE TITLE
Simple Server / Using instructions for generation do not work

### ISSUE THREAD

**Issue by halostatue** [hofstadter-io/hof-docs#5]
https://docs.hofstadter.io/first-example/simple-server/using/ has an instruction to run `hof gen ./example`, which fails.

```console
❯ hof gen ./example
LoadCue: import failed: cannot find package "hof.io/docs/first-example/gen" true []
import failed: cannot find package "hof.io/docs/first-example/gen":
    ./example/gen.cue:4:2
```

This may be related to https://docs.hofstadter.io/first-example/simple-server/setup/ instructions on CUE module creation:

```
hof mod init cue hof.io/example
```

Changing the instruction to

```
hof mod init cue hof.io/first-example
```

Also does not work.

Changing the instruction to

```
hof mod init cue hof.io/docs/first-example
```

Results in an error in a different location:

```console
❯ hof gen ./example
LoadCue: import failed: import failed: cannot find package "github.com/hofstadter-io/hof/schema/gen" true []
import failed: import failed: cannot find package "github.com/hofstadter-io/hof/schema/gen":
    ./example/gen.cue:4:2
    ./gen/server.cue:4:2
```

This *may* be related to an incomplete configuration in `cue.mods` (comparing the two folders):

```diff
--- hof-first-example/cue.mods	2022-06-06 16:36:59.000000000 -0400
+++ ./cue.mods	2022-06-06 16:39:04.000000000 -0400
@@ -1,3 +1,7 @@
 module hof.io/docs/first-example

-cue v0.4.0
+cue v0.4.1
+
+require (
+	github.com/hofstadter-io/hof v0.6.1
+)
```

**Comment by halostatue** [comment]
The same error shows up in a local clone of `hof-docs/code/first-example/simple-server`. I am stuck moving forward in the tutorial / worked first example from this point.

**Comment by verdverm** [comment]
Thanks for the report @halostatue, I'll try to go through them myself.

Each folder is trying to relate to the section, such that new sections start at the completion of the previous section. It may be that I fixed things in later sections and did not backport them to previous sections.

If you are interested, I can schedule a time to stream this live on [Twitch](https://twitch.tv/dr_verm) this week. That is where I typically proofread the docs

**Comment by halostatue** [comment]
I’m mostly interested in going through the worked first example to make sure that I can understand what `hof` is doing to see if it’s appropriate for something my team is building. This isn’t *urgent* for us, so I’ll just keep an eye on progress via this ticket, if that’s OK.

**Comment by verdverm** [comment]
sure, I'll let you know once I have made the updates. Thanks for the reports and your patience

**Comment by verdverm** [comment]
btw, I think you need to run `hof mod vendor cue` to fetch the missing dependency, in addition to the fixes you have above

**Comment by verdverm** [comment]
_dev note_: bump multiple versions when making the fixes and updates

**Comment by halostatue** [comment]
`hof mod vendor cue` got me further along. Now I get the following error:

```console
❯ hof gen ./example -s
Server.In.SERVER.Port: cannot convert non-concrete value >1024 & int (and 2 more errors)

Errors while loading generators
```

Once I removed the `Port: int & >1024` definition (and a couple of other items that differed from the `hof-docs/code/…` example), I was able to move forward.

I haven’t moved forward on this tonight, as I have other things to do tonight. I’ll get back to it later this week.
